docs/ref/forms/renderers.txt

```diff
@@ -0,0 +1,117 @@
+======================
+The form rendering API
+======================
+
+.. module:: django.forms.renderers
+   :synopsis: Built-in form renderers.
+
+.. versionadded:: 1.11
+
+Django form widgets are rendered using the Django template backend system.
+
+The form rendering process can be customized at several levels:
+
+* Widgets can specify custom template names.
+* Forms and widgets can specify custom renderer classes.
+* Built-in templates can be overridden within a project or application.
+
+.. _low-level-widget-render-api:
+
+The low-level render API
+========================
+
+Widget templates are stored in the ``django/forms/widgets`` path. You can
+provide a custom template for ``input.html`` by defining
+``django/forms/widgets/input.html``, for example.
+
+The rendering of form templates is controlled by a customizable renderer class.
+A custom renderer can be specified by updating the :setting:`FORM_RENDERER`
+setting. It defaults to
+``'``:class:`django.forms.renderers.DjangoTemplateRenderer`\ ``'``.
+
+Use one of the :ref:`built-in template form renderers
+<built-in-template-form-renderers>` or implement your own. Custom renderers
+must implement a ``render(template_name, context, request=None)`` method. It
+should return a :class:`~django.template.Template` object or raise
+:exc:`~django.template.TemplateDoesNotExist`.
+
+.. _built-in-template-form-renderers:
+
+Built-in-template form renderers
+================================
+
+``DjangoTemplateRenderer``
+--------------------------
+
+.. class:: DjangoTemplateRenderer
+
+This renderer uses a :class:`~django.template.backends.django.DjangoTemplates`
+engine. It loads templates first from the application directories and then from
+the built-in form templates directory in ``django/forms``.
+
+``Jinja2TemplateRenderer``
+--------------------------
+
+.. class:: Jinja2TemplateRenderer
+
+This renderer is the same as :class:`DjangoTemplateRenderer` except that it
+uses a :class:`~django.template.backends.jinja2.Jinja2` backend.
+
+To use this backend, all the widgets in your project and its third-party apps
+must have Jinja2 templates. You can't use this renderer, for example, if you're
+using :mod:`django.contrib.admin` because Jinja2 templates aren't included for
```

Comment:
Is it worth adding "(unless you provide Jinja templates for all the admin widgets yourself)" here?